DietPi-Software | Samba: Move disk cache to RAM

MichaIng opened the pull request:
**Status**: Ready
- [x] Changelog

**Testing**:
- [x] Install
- [x] Service
- [x] Reboot
- [x] Reinstall
- [x] Uninstall

🈯️ Passed on Jessie, Stretch and Buster

**Reference**: https://github.com/Fourdee/DietPi/issues/2396

MichaIng committed:
v6.20

+ DietPi-Software | Samba: Move disk cache to RAM

MichaIng committed:
v6.20

+ CHANGELOG | Samba: Moved disk cache to RAM to reduce disk I/O
+ CHANGELOG | Removed obsolete fake-hwclock entry

Fourdee commented on dietpi/dietpi-software:
```diff
@@ -3344,6 +3344,17 @@ _EOF_
+			# - Link disk cache to RAM
+			ln -s /var/run/samba-cache /var/cache/samba
+			# - Create RAM cache dir automatically on boot
+			echo 'd /var/run/samba-cache - - - - -' > /etc/tmpfiles.d/dietpi-samba_cache.conf
```
@MichaIng 

Nice 👍 

Reading this, does it actually make a tmpfs file, as ```/var``` is mounted to disk?
https://developers.redhat.com/blog/2016/09/20/managing-temporary-files-with-systemd-tmpfiles-on-rhel7/

If not, we might be better off using ```/tmp```?

WolfganP replied:
/var/run is linked to /run which is a tmpfs already.

Fourdee replied:
@WolfganP 

Legend 👍 💘 
```
root@Fuzon:~# ls -lha /var/run
lrwxrwxrwx 1 root root 4 Jul 21 21:56 /var/run -> /run
```

MichaIng replied:
@Fourdee 
Jep, `tmpfiles` has nothing to do with `tmpfs` (RAMdisk). But as stated, /var/run -> /run is tmpfs as well.

Fourdee commented:
@MichaIng @WolfganP 

Looks good from my end, nice implementation. @WolfganP ?

WolfganP commented:
> Looks good from my end, nice implementation. @WolfganP ?

@Fourdee @MichaIng LGTM, couldn't test on my main system as is not migrated yet, but I tested manually and worked OK

Commit:
v6.20

+ Reinstall samba during patch: https://github.com/Fourdee/DietPi/issues/2396#issuecomment-451701569
